motion/colav/scripts/VO_controller_node.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python3
+
+from velocity_obstacle_node import VelocityObstacle
+import rospy
+from nav_msgs.msg import Odometry
+from std_srvs.srv import Trigger,TriggerResponse
+from vortex_msgs.msg import GuidanceData
+from tf.transformations import euler_from_quaternion
+
+
+
+class VOControllerNode:
+
+    """
+    The velocity object controller node.
+    Changes the ASV velocity such that a collision cannot occur.
+
+    """
+    
+
+    def __init__(self):
+        rospy.init_node("colav",anonymous=True)
+
+        self.vessel = Odometry()
+        self.obstacle = Odometry()
+
+
+        # Subscribers
+        self.vessel_sub = rospy.Subscriber(
+            "/pose_gt", Odometry, self.vessel_callback, queue_size=1
+        )  # 20hz
+
+        self.obstacle_sub = rospy.Subscriber(
+            "/obstacle_gt", Odometry, self.obstacle_callback, queue_size=1
+        )  # 20hz
+
+        # Service
+        self.colav_srv = rospy.Service("/colav",Trigger,self.avoid_collision)
+        
+        self.velocity_pub = rospy.Publisher("/guidance/los_data",GuidanceData,queue_size=1)
+    
+
+
+
+    def avoid_collision(self,trigger):
+
+
+        """
+
+        Avoids collison by continuously updating the desired velociy,
+        and sending it to the AUV's velocity controller. This is implemented as a service that 
```

Comment:
We do not have a velocity controller, but we have a surge controller (named speed controller), and also a heading controller